com.unity.uiwidgets/Editor/UIWidgetsEditorPanel.cs

```diff
@@ -48,15 +50,39 @@ public class UIWidgetsEditorPanel : EditorWindow, IUIWidgetsWindow {
             D.assert(_wrapper != null);
             _wrapper?.Destroy();
             _wrapper = null;
-
+            _internalTextFonts.Clear();
             Input_OnDisable();
         }
+        
+        protected virtual void loadConfiguration() {}
+
+        Dictionary<string,UIWidgetsPanel.TextFont> _internalTextFonts = new Dictionary<string, UIWidgetsPanel.TextFont>();
+        protected void AddFont(string family, List<string> assets, List<int> weights) {
+            if (assets.Count != weights.Count) {
+                UnityEngine.Debug.Log($"The size of {family}‘s assets should be equal to the weights'.");
+                return;
+            }
+            UIWidgetsPanel.TextFont textFont = new UIWidgetsPanel.TextFont();
+            textFont.family = family;
+            UIWidgetsPanel.Font[] fonts = new UIWidgetsPanel.Font[assets.Count];
+            for (int j = 0; j < assets.Count; j++) {
+                UIWidgetsPanel.Font font = new UIWidgetsPanel.Font();
+                font.asset = assets[j];
+                font.weight = weights[j];
+                fonts[j] = font;
+            }
+            textFont.fonts = fonts;
+            _internalTextFonts.Add(family,textFont);
+        }
 
         void OnEnable() {
             D.assert(_wrapper == null);
             _wrapper = new UIWidgetsPanelWrapper();
-            _wrapper.Initiate(this, _currentWidth, _currentHeight, _currentDevicePixelRatio, new Dictionary<string, object>());
-            
+            loadConfiguration();
+            var settings = new Dictionary<string, object>();
+            UIWidgetsPanel.TextFont[] textFonts = loadTextFont();
+            settings.Add("fonts", _wrapper.fontsToObject(textFonts));
+            _wrapper.Initiate(this, _currentWidth, _currentHeight, _currentDevicePixelRatio, settings);
```

Comment:
can we pass the raw _internalTextFonts to _wrapper.Initiate like:
_wrapper.Initiate(this, _currentWidth, _currentHeight, _currentDevicePixelRatio, _internalTextFonts);

because _wrapper only requires the raw font data here, we can leave all the post processes like loadTextFont() and fontsToObject inside the _wrapper.Initiate function

ps, loadTextFont is not necessary here, we can pass _internalTExtFonts to fontsToObject() directly